packages/DropdownMenu/src/DropdownMenu.js

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Popover from "@paprika/popover";
+import ContentContainerStyled from "./ContentContainer.styles";
+import Confirmation from "./Confirmation/Confirmation";
+import Divider from "./Divider/Divider";
+import Trigger from "./Trigger/Trigger";
+import Item from "./Item/Item";
+
+const { alignTypes, oneOf, node, string, func } = PropTypes;
+
+const propTypes = {
+  /** Alignment of the dropdown menu */
+  align: oneOf(alignTypes),
+
+  /** Children should consist of <Dropdown.Item /> */
+  children: node.isRequired,
+
+  /** Optional custom classes */
+  className: string,
+
+  /** Render prop for rendering the trigger element that toggles the dropdown */
+  renderTrigger: func.isRequired,
+};
+
+const defaultProps = {
+  align: "bottom",
+  className: "",
+};
+
+const DropdownMenu = props => {
+  const { align, children, className } = props;
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isConfirming, setIsConfirming] = React.useState(false);
+  const [renderConfirmation, setRenderConfirmation] = React.useState(null);
+
+  const handleCloseMenu = () => {
+    setIsOpen(false);
+    setIsConfirming(false);
+    setRenderConfirmation(null);
+  };
+
+  const handleOpenMenu = () => {
+    setIsOpen(true);
+  };
+
+  const getTriggerStateAndHelpers = () => {
+    return {
+      isOpen,
+      handleOpenMenu,
+    };
+  };
+
+  const handleShowConfirmation = renderConfirmation => () => {
+    setIsConfirming(prevIsConfirmingState => !prevIsConfirmingState);
+    setRenderConfirmation(prevIsConfirmingState => (prevIsConfirmingState ? null : renderConfirmation));
+  };
+
+  const renderContent = () => {
+    if (isConfirming) {
+      return renderConfirmation(handleCloseMenu);
+    }
+
+    return (
+      <ContentContainerStyled isOpen={isOpen}>
+        {React.Children.toArray(children).map(child => {
+          if (child.type.componentType === "DropdownMenu.Item") {
+            if (child.props.renderConfirmation) {
+              return React.cloneElement(child, {
+                onShowConfirmation: handleShowConfirmation(child.props.renderConfirmation),
+              });
+            }
+            return React.cloneElement(child, {
+              onClose: handleCloseMenu,
+            });
+          }
+
+          return child;
+        })}
+      </ContentContainerStyled>
+    );
+  };
+
+  return (
+    <Popover
+      align={align}
+      className={className}
+      offset={4}
+      role={!isConfirming ? "menu" : null}
+      isOpen={isOpen}
+      onClose={() => {
+        if (!isConfirming) {
+          handleCloseMenu();
+        }
+      }}
+    >
+      <Popover.Trigger>{props.renderTrigger(getTriggerStateAndHelpers())}</Popover.Trigger>
```

Comment:
In the storybook you'll notice that the Trigger Button gets the focus twice... because a RawButton is being wrapped around the Trigger.  I think this is a bit of a flaw in our Popover API, but for now, if you pass a function as the `children`, it won't do that, and the `A11y` will be better.  So, simply:
```jsx
<Popover.Trigger>{() => props.renderTrigger(getTriggerStateAndHelpers())}</Popover.Trigger>
```
Will remedy this (for now).

Reply:
(You could leave a `// TODO:` in the code to fix this later though)

Reply:
Can you created an issue about this, so we can address this later on the popover @tristanjasper 🙏

Reply:
I can create the issue... I noticed it first 😬